packages/form-core/src/FormApi.ts

```diff
@@ -747,6 +747,26 @@ export class FormApi<
               }))
             }
           }
+        } else {
```

Comment:
Here the form validator should clear the field errors only if the validator is one that returns the object version of `FormValidationError`.
This means you should go with `else if (isFormValidationError(rawError))`.

However, when the form has no errors the adapters are returning `undefined`. With this information alone you don't know if it was a form object validator or a simple validator returning a string, as in that case you shouldn't clear fields (the new failing tests you're getting) but only the form error.

The first approach that comes to my mind is to slightly refactor the adapters and make them return something instead of undefined. It could be `{ fields: {} }`

With this information you know that the form validator returned no errors and you should clear the fields.

Note: this might break the initial `if (fieldErrors)` but something like `if (Object.values(fieldErrors).length)` should do the trick.